### PR TITLE
FEAT: Implement a ground and the collision with the player second game

### PR DESCRIPTION
**Pull request by Thomaltarix**
I implemented the ground (which is only a tile when the player can stay).

Obviously, if the player can stay on it, I implemented collisions.

I also made jumps and gravity a bit more realistic by changing some values.

> [!Note]
> For the moment, the player can press space bar and fly, so I thought about three methods to fix it.

The first one:
- Implement a system of jetpack, and stamina for the jump, like you can jump while you have stamina, when you passively recover it when not pressing space bar. But when you're out of stamina, you can't jump anymore while your bar isn't full. (I almost fully know how to implement it)

The second one:
- When pressing space bar, it does a real jump, for a fixed amount of seconds, a fixed high, etc... But we'll need to implement a clock or something, and I don't know yet how to do it.

Third and last one:
- Keep it like this, and assume that we didn't implement a real jump for the moment, and our game is a Beta.

> [!Important]
> Tell me which option do you prefer please ! (So that I can adapt my self or ask someone do to it for me)

> [!Note]
> I leaved the display of hitbox for now to help debugging a bit. (Since this isn't merged to main, we'll just need to remove it when the game will be release to main) (Just removed this line in `firstScene.json` 
image

### CODE REVIEW

**Review by Peralban — APPROVED** [review]

File: QuentinTheFarmerGame/Systems/Menus/Buttons/SingleplayerButton/SingleplayerButtonInitSystem.cpp [hunk 1]
```diff
@@ -18,9 +18,6 @@ SingleplayerButtonInitSystem::SingleplayerButtonInitSystem() :
 
 static void handleThis(ECS::Registry& reg, int idxPacketEntities)
 {
-    try {
```
Comment (Thomaltarix) [reply QuentinTheFarmerGame/Systems/Menus/Buttons/SingleplayerButton/SingleplayerButtonInitSystem.cpp:21]: I'll copy paste the R-Type one, mb

Comment (mathieurobert1) [inline QuentinTheFarmerGame/Systems/Menus/Buttons/SingleplayerButton/SingleplayerButtonInitSystem.cpp:21]: You don't start a Single Player mode ?

**Review by mathieurobert1 — APPROVED** [review]
I think for the jump problem, you can just stop to jump when your player are not touching the ground.
I saw a hitbox collision system, I think you can use it to know if the player touch the ground when he jump.

File: QuentinTheFarmerGame/Systems/Menus/Buttons/SingleplayerButton/SingleplayerButtonInitSystem.cpp (hunk 1 above)
Comment (Thomaltarix) [reply QuentinTheFarmerGame/Systems/Menus/Buttons/SingleplayerButton/SingleplayerButtonInitSystem.cpp:21]: Done

**Pull request merged**